Sources/PackageModel/PackageReference.swift

```diff
@@ -173,6 +173,12 @@ extension PackageReference: Hashable {
     }
 }
 
+extension PackageReference {
+    public var diagnosticsMetadata: ObservabilityMetadata {
+        return .packageMetadata(identity: self.identity, kind: self.kind)
+    }
+}
```

Comment:
could this be a non-public extension closer to the call site? seems fairly leaky as a public property of the model